src/GusApi/Client/SoapClient.php

```diff
@@ -4,9 +4,12 @@
 
 class SoapClient extends \SoapClient
 {
-    public function __doRequest($request, $location, $action, $version, $one_way = 0): string
+    public function __doRequest($request, $location, $action, $version, $oneWay = false): string
     {
-        $response = parent::__doRequest($request, $location, $action, $version, $one_way);
+        $response = parent::__doRequest($request, $location, $action, $version, $oneWay);
+        if (null === $response) {
+            return $this->__doRequest($request, $location, $action, $version, $oneWay);
```

Comment:
This may cause an infinite loop. It's better to throw an exception instead and let the application decide what to do next. @johnzuk I'd suggest to add `'exceptions' => true` to default options here: https://github.com/johnzuk/GusApi/blob/a9f777f046fbb8d28c482cb6b0b00e77738d70a5/src/GusApi/Client/Builder.php#L53 but that's a breaking change I guess.

Reply:
For sure it can be like what - exception sound like best option here

Reply:
@bmxmale Also additional logic needs test coverage